subprojects/core/src/main/java/org/gradle/api/internal/provider/DefaultPropertyState.java

```diff
@@ -116,6 +116,7 @@ public boolean isPresent() {
 
     @Override
     public String toString() {
-        return String.format("value: %s", getOrNull());
+        // NOTE: This should not realize the Provider since the debugger will try to call this method and make debugging really frustrating.
```

Comment:
This comment is really confusing.  You can read it one way as "the current behavior is that this does not realize the provider and everything is ok, so don't change it" or you can read it another way that "the current behavior is that this realizes the Provider and it really shouldn't, so we should fix this".  We should disambiguate the language here.